test/plugins/stats.js

```diff
@@ -0,0 +1,54 @@
+const request = require('supertest')
+const express = require('express')
+const {createRobot} = require('../..')
+const plugin = require('../../lib/plugins/stats')
+const nock = require('nock');
+
+nock.disableNetConnect();
+nock.enableNetConnect(/127\.0\.0\.1/);
+
+describe('stats', function () {
+  let robot
+  let server, github
+
+  beforeEach(async () => {
+    nock('https://api.github.com')
+     .defaultReplyHeaders({'Content-Type': 'application/json'})
+     .get('/app/installations?per_page=100').reply(200, [{id: 1, account: {login: 'testing'}}])
+     .get('/installation/repositories').reply(200, {repositories: [
+       {private: true, stargazers_count: 1},
+       {private: false, stargazers_count: 2}
+     ]})
+
+    const cacheManager = require('cache-manager')
+    const cache = cacheManager.caching({store: 'memory'})
+
+    const GitHubApi = require('github')
+    const app = {
+      async asApp() {
+        return new GitHubApi()
+      },
+
+      async asInstallation() {
+        return new GitHubApi()
+      },
+
+      async createToken() {
+        return {data: {token: 'test'}}
+      }
+    }
+    robot = createRobot({app, cache})
```

Comment:
This all needs cleaned up, but I'd prefer to do that in a separate pull request that is focused on updating how we test (to make it easier to test with nock instead of mocking the github client).

I'd like to have some kind of `probot-test` package that includes a test harness for setting up `robot`.